crates/net/eth-wire-types/src/status.rs

```diff
@@ -377,6 +393,66 @@ impl Encodable for StatusMessage {
     }
 }
 
+impl Display for StatusMessage {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            StatusMessage::Legacy(s) => Display::fmt(s, f),
+            StatusMessage::Eth69(s69) => Display::fmt(s69, f),
+        }
+    }
+}
+
+impl From<UnifiedStatus> for StatusMessage {
+    fn from(s: UnifiedStatus) -> Self {
+        if s.version == EthVersion::Eth69 {
```

Comment:
can we move the impls of these from< impls into functions of UnifiedStatus (`into_legacy` + `into_status69`) this makes it easier to find these and obvious which conversion is called

Reply:
@mattsse, I feel you on this. nice to have all the conversion logic encapsulated in the `UnifiedStatus` impl. updated in latest commit https://github.com/paradigmxyz/reth/pull/16099/commits/19828ede5bc0c95108e27bcf0a9d69db5f407fc2

will start on the tests tomorrow 🤠

Reply:
@mattsse, updated the test coverage to support `UnifiedStatus` + resolved linting in latest commit https://github.com/paradigmxyz/reth/pull/16099/commits/edd85b3ecfcb57288a2afd348539f5eb1c14e7c6